### ISSUE TITLE
Add missing reagent style

### ISSUE THREAD

**Issue by sirmspencer** [borkdude/clj-kondo#715]
1. I have noticed a case that clj-kondo does not interpret the Reagent code correctly.  reagent/with-let cannot tell that the following form is the same as a let. 

In the following example, a is flagged as 'unresolved symbol'
`(defn example []    
  (r/with-let [a "test"]    
    [:div ...]))`

The correct behavior should be no warnings.

2. Another case is clj-kondo considers a parameter missing when its not used in the did-mount section before the anonymous render function, but it should be considered used if it is used in the did-mount section OR the anonymous function.  

In the example below clj-condo will say that b is unused, when clearly it is used in the anonymous function.
`(defn example [a b]    
  (do-something a)    
    (fn [a b]    
      [:div ...])))`

Doc for form-2: https://github.com/reagent-project/reagent/blob/master/doc/CreatingReagentComponents.md#form-2--a-function-returning-a-function

The correct behavior should be no warnings.

**Comment by borkdude** [comment]
@sirmspencer For this you can use `:lint-as`:

```
(ns foo
  {:clj-kondo/config '{:lint-as {reagent.core/with-let clojure.core/let}}}
  (:require [reagent.core :as r]))

(defn example [] (r/with-let [a "test"] [:div a]))
```

This gives no warnings. Note that the config should go into `.clj-kondo/config.edn`.

In the case of a form-2 component, you should just rename the unused outer parameter to `_b` to mark it as used.

**Comment by borkdude** [comment]
Also see: 

- https://github.com/borkdude/clj-kondo/blob/master/doc/config.md#lint-a-custom-macro-like-a-built-in-macro
- https://github.com/borkdude/clj-kondo/blob/master/doc/config.md#exclude-unused-bindings-from-being-reported

**Comment by sirmspencer** [comment]
Thank you.  I apparently need to read through the config guide again.

**Comment by borkdude** [comment]
No worries! There's also a #clj-kondo Slack channel for quick questions.
